functions/Copy-SqlSsisCatalog.ps1

```diff
@@ -0,0 +1,472 @@
+                foreach ($p in $projs) {
+                    $remove.Projects[$p].Drop()
+                }
+                $remove.Drop()
+                $destinationCatalog.Alter()
+                $destinationCatalog.Refresh()
+            }
+            Write-Output "Creating folder $Folder."
+            $destFolder = New-Object "$ISNamespace.CatalogFolder" ($destinationCatalog, $Folder, $Description)
+            $destFolder.Create()
+            $destFolder.Alter()
+            $destFolder.Refresh()
+        }
+        
+        Function New-FolderEnvironment {
+            param(
+                [String]$Folder,
+                [String]$Environment,
+                [Switch]$Force
+            )
+            $envDestFolder = $destinationFolders | Where-Object { $_.Name -eq $Folder }
+            if ($force) {
+                $envDestFolder.Environments[$Environment].Drop()
+                $envDestFolder.Alter()
+                $envDestFolder.Refresh()
+            }
+            $srcEnv = ($sourceFolders | Where-Object { $_.Name -eq $Folder }).Environments[$Environment]
+            $targetEnv = New-Object "$ISNamespace.EnvironmentInfo" ($envDestFolder, $srcEnv.Name, $srcEnv.Description)
+            foreach ($var in $srcEnv.Variables) {
+                if ($var.Value.ToString() -eq "") { 
+                    $finalValue= ""
+                }
+                else {
+                    $finalValue = $var.Value
+                }
+                $targetEnv.Variables.Add($var.Name, $var.Type, $finalValue, $var.Sensitive, $var.Description)
+            }
+            Write-Output "Creating environment $Environment."
+            $targetEnv.Create()
+            $targetEnv.Alter()
+            $targetEnv.Refresh()
+        }
+
+        Function New-SSISDBCatalog {
+            param(
+                [System.Security.SecureString]$Password
+            )
+            if(!$Password) {
+                Write-Output "SSISDB Catalog requires a password."
+                $pass1 = Read-Host "Enter a password" -AsSecureString
+                $plainTextPass1 = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($pass1))
+                $pass2 = Read-Host "Re-enter password" -AsSecureString
+                $plainTextPass2 = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($pass2))
+                if ($plainTextPass1 -ne $plainTextPass2) {
+                    throw "Validation error, passwords entered do not match."
+                }
+                $plainTextPass = $plainTextPass1
+            }
+            else {
+                $plainTextPass = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($Password))
+            }
+
+            $catalog = New-Object "$ISNamespace.Catalog" ($destinationSSIS, "SSISDB", $plainTextPass)  
+            $catalog.Create()
```

Comment:
Interesting - you must first change to enable clr if it's not enabled. Please prompt to confirm.
Exception calling "Create" with "0" argument(s): "Integration Services uses CLR stored procedures. Enable "clr enabled" configuration option."
At C:\github\branches\dbatools\functions\Copy-SqlSsisCatalog.ps1:207 char:13
+             $catalog.Create()
+             ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : IntegrationServicesException